pyanaconda/payload/install_tree_metadata.py

```diff
@@ -0,0 +1,244 @@
+        try:
+            result = self._session.get("%s/%s" % (url, file_name), headers=headers,
+                                       proxies=proxies, verify=verify)
+            # Server returned HTTP 4XX or 5XX codes
+            if result.status_code >= 400 and result.status_code < 600:
+                log.info("Server returned %i code", result.status_code)
+                return (None, result.status_code)
+            log.debug("Retrieved '%s' from %s", file_name, url)
+        except requests.exceptions.RequestException as e:
+            log.info("Error downloading '%s': %s", file_name, e)
+            return (None, None)
+        return (result, result.status_code)
+
+    def get_release_version(self):
+        """Get release version from the repository.
+
+        :returns: Version as lowercase string.
+        :raises: ValueError if version is not present.
+        """
+        version = self._tree_info.release.version
+
+        if not version:
+            raise ValueError("Can't read release version from the .treeinfo file! "
+                             "Is the .treeinfo file corrupted?")
+
+        return version.lower()
+
+    def get_metadata_repos(self):
+        """Get all repository metadata objects."""
+        self._check_variants()
+
+        return self._meta_repos
+
+    def get_repo_metadata_by_name(self, name):
+        """Get repository metadata object with given name.
+
+        :param name: Name of the variant to return.
+        :rtype name: VariantRepo object or None.
+        """
+        self._check_variants()
+
+        for variant in self._meta_repos:
+            if name == variant.name:
+                return variant
+
+    def get_base_repo_metadata(self, additional_names=None):
+        """Get repo metadata about base repository."""
+        self._check_variants()
+        repos = constants.DEFAULT_REPOS
+
+        if additional_names:
+            repos.extend(additional_names)
+
+        for repo_md in self._meta_repos:
+            if repo_md.name in repos:
+                return repo_md
+
+        return None
+
+    def _check_variants(self):
+        if not self._meta_repos:
+            self._read_variants()
+
+    def _read_variants(self):
+        for variant_name in self._tree_info.variants:
+            variant_object = self._tree_info.variants[variant_name]
+            self._meta_repos.append(RepoMetadata(variant_name, variant_object, self._path))
+
+
+class RepoMetadata(object):
+    """Metadata repo object contains metadata about repository."""
+
+    def __init__(self, name, obj, root_path):
+        """Do not instantiate this class directly.
+        Use InstallTreeMetadata to instantiate this class.
+        """
+        self._name = name
+        self._obj = obj
+        self._root_path = root_path
+
+    @property
+    def name(self):
+        """Get name of this repository."""
+        return self._name
+
+    @property
+    def repo_rel_path(self):
```

Comment:
Is the `repo` in the name necessary? Also, please don't use shortcuts. Also, we don't use this, so it might be hidden.

Reply:
Name changed. I don't see point in hiding that. It can be used in the future and it is used by member methods now.